mobile/CmpE451_App/src/screen/CreatePost.js

```diff
@@ -212,125 +108,54 @@ export default function CreatePost({route}) {
           maxHeight: '50%',
           alignItems: 'flex-start',
         }}>
-        {index ? (
-          <View>
+        {textInputs.map((input, key) => (
+          <View style={styles.inputContainer}>
             <TextInput
-              placeholder="Add a title"
-              placeholderTextColor="black"
-              style={{alignSelf: 'flex-start', marginLeft: 10, fontSize: 20}}
+              placeholder={input.key}
+              value={input.value}
+              onChangeText={text => {
+                inputHandler(text, key, input.key);
+              }}
             />
           </View>
-        ) : (
-          <ScrollView showsVerticalScrollIndicator={false}>
-            {myForum()}
-          </ScrollView>
-        )}
-      </View>
-      <View
-        style={{
-          marginTop: 20,
-          borderTopLeftRadius: 8,
-          borderTopRightRadius: 8,
-          borderTopColor: 'grey',
-          borderTopWidth: 0.3,
-          height: '40%',
-          elevation: 6,
-          backgroundColor: 'white',
-          alignContent: 'center',
-        }}>
-        <View style={{alignContent: 'center'}}>
-          <Text style={{alignSelf: 'center'}}>What Do You Want To Post?</Text>
-          <View
-            style={{
-              width: '100%',
-              justifyContent: 'space-evenly',
-              padding: 5,
-            }}>
-            <View
-              style={{
-                flexDirection: 'row',
-                justifyContent: 'space-evenly',
-                padding: 5,
-              }}>
-              <View style={{}}>
-                <TouchableOpacity onPress={() => setIndex(index + 1)}>
-                  <Icon name="ios-text" style={{}} size={30}></Icon>
-                  <Text style={{size: '3'}}>Text</Text>
-                </TouchableOpacity>
-              </View>
-              <View style={{}}>
-                <Icon name="image-outline" style={{}} size={30}></Icon>
-                <Text style={{size: '3'}}>Audio</Text>
-              </View>
-              <View style={{}}>
-                <Icon name="link" style={{}} size={30}></Icon>
-                <Text style={{size: '3'}}>Link</Text>
+        ))}
+        {linkedFieldNames[0].value != ''
```

Comment:
```{linkedFieldNames[0].value != ''``` not working. 
 The fields started to appear when I removed it. 
You should do the control another way.